File: algorithms/200-299/240. Search a 2D Matrix II.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        if (matrix.empty()) return false;
        int n = matrix.size(), m = matrix[0].size();
        int i = 0, j = m -1;
        while (i <= n-1 && j >= 0){
            int num = matrix[i][j];
            if (num == target){
                return true;
            } else if (num > target){
                j -= 1;
            } else{
                i += 1;
            }
        }
        return false;
    }
};

File: algorithms/200-299/240. Search a 2D Matrix II.py
class Solution:
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        if not matrix:
            return False
        n, m = len(matrix), len(matrix[0])
        i, j = 0, m - 1
        while i <= n - 1 and j >= 0:
            num = matrix[i][j]
            if num == target:
                return True
            elif num > target:
                j -= 1
            else:
                i += 1
        return False
        